code/game/dna/genes/vg_powers.dm

```diff
@@ -37,70 +37,101 @@
-		hairs.Add(H.name) // add hair name to hairs
-		qdel(H) // delete the hair after it's all done
-
-	var/new_style = input("Please select hair style", "Character Generation",M.h_style)  as null|anything in hairs
+	var/list/valid_hairstyles = M.generate_valid_hairstyles()
+	var/new_style = input("Please select hair style", "Character Generation", head_organ.h_style) as null|anything in valid_hairstyles
 
 	// if new style selected (not cancel)
 	if (new_style)
-		M.h_style = new_style
-
-	// facial hair
-	var/list/all_fhairs = subtypesof(/datum/sprite_accessory/facial_hair)
-	var/list/fhairs = list()
+		head_organ.h_style = new_style
 
-	for(var/x in all_fhairs)
-		var/datum/sprite_accessory/facial_hair/H = new x
-		fhairs.Add(H.name)
-		qdel(H)
+	var/new_hair = input("Please select hair color.", "Character Generation", rgb(head_organ.r_hair, head_organ.g_hair, head_organ.b_hair)) as null|color
+	if(new_hair)
+		head_organ.r_hair = hex2num(copytext(new_hair, 2, 4))
+		head_organ.g_hair = hex2num(copytext(new_hair, 4, 6))
+		head_organ.b_hair = hex2num(copytext(new_hair, 6, 8))
 
-	new_style = input("Please select facial style", "Character Generation",M.f_style)  as null|anything in fhairs
+	// facial hair
+	var/list/valid_facial_hairstyles = M.generate_valid_facial_hairstyles()
+	new_style = input("Please select facial style", "Character Generation", head_organ.f_style) as null|anything in valid_facial_hairstyles
 
 	if(new_style)
-		M.f_style = new_style
+		head_organ.f_style = new_style
 
-	var/new_gender = alert(usr, "Please select gender.", "Character Generation", "Male", "Female")
-	if (new_gender)
-		if(new_gender == "Male")
-			M.change_gender(MALE)
-		else
-			M.change_gender(FEMALE)
+	var/new_facial = input("Please select facial hair color.", "Character Generation", rgb(head_organ.r_facial, head_organ.g_facial, head_organ.b_facial)) as null|color
+	if(new_facial)
+		head_organ.r_facial = hex2num(copytext(new_facial, 2, 4))
+		head_organ.g_facial = hex2num(copytext(new_facial, 4, 6))
+		head_organ.b_facial = hex2num(copytext(new_facial, 6, 8))
+
+	//Head accessory.
+	if(head_organ.species.bodyflags & HAS_HEAD_ACCESSORY)
+		var/list/valid_head_accessories = M.generate_valid_head_accessories()
+		var/new_head_accessory = input("Please select head accessory style", "Character Generation", head_organ.ha_style) as null|anything in valid_head_accessories
+		if(new_head_accessory)
+			head_organ.ha_style = new_head_accessory
+
+		var/new_head_accessory_colour = input("Please select head accessory colour.", "Character Generation", rgb(head_organ.r_headacc, head_organ.g_headacc, head_organ.b_headacc)) as null|color
+		if(new_head_accessory_colour)
+			head_organ.r_headacc = hex2num(copytext(new_head_accessory_colour, 2, 4))
+			head_organ.g_headacc = hex2num(copytext(new_head_accessory_colour, 4, 6))
+			head_organ.b_headacc = hex2num(copytext(new_head_accessory_colour, 6, 8))
+
+	//Body markings.
+	if(M.species.bodyflags & HAS_MARKINGS)
```

Comment:
Means that if you aren't a species that has markings, this option won't pop up when you use the morph ability.